File: adventofcode/2024/02_1.cpp
#include <bits/stdc++.h>

using namespace std;

bool solve(vector<int> &v) {
    bool inc = v[0] < v[1];
    for(int i=0; i<v.size()-1; i++) {
        if ((inc && v[i] >= v[i+1]) || (!inc && v[i] < v[i+1])) return false;
        if (v[i] == v[i+1] || abs(v[i]-v[i+1]) > 3) return false;
    }
    return true;
}

signed main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    #ifndef ONLINE_JUDGE 
        freopen("../input.txt", "r", stdin); 
        freopen("../output.txt", "w", stdout); 
    #endif
    const int LINES=1000;
    int ans = 0;
    for(int i=0; i<LINES; i++)
    {
        string s; getline(cin, s);
        stringstream ss(s);
        vector<int> v;
        int a;
        while (ss >> a) {
            v.push_back(a);
        }
        if (solve(v)) ans++;
    }
    cout << ans;
    return 0;
}

File: adventofcode/2024/python/02_1.py
import sys

def solve(v):
    inc = v[0] < v[1]
    for i in range (len(v)-1):
        if (inc and v[i] >= v[i+1]) or (not inc and v[i] < v[i+1]):
            return False
        if v[i] == v[i+1] or abs(v[i]-v[i+1]) > 3:
            return False
    return True

def main():
    sys.stdin = open("../input.txt", "r")
    sys.stdout = open("../output.txt", "w")
    
    LINES = 1000
    ans = 0
    
    for _ in range(LINES):
        try:
            s = input().strip()
        except EOFError:
            break
        if not s:
            continue
        v = list(map(int, s.split()))
        if solve(v):
            ans+=1
    
    print(ans)

if __name__ == "__main__":
    main()